Dockerfile

```
@@ -1,16 +1,28 @@
-FROM debian:bookworm-slim
+FROM python:3.11-slim-trixie AS joinmarket
 
-RUN mkdir -p /jm/clientserver
 WORKDIR /jm/clientserver
+COPY ./pubkeys ./pubkeys
+COPY ./install.sh ./install.sh
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends \
+    gnupg \
+    ca-certificates \
+    curl \
+  && ./install.sh --docker-install --deps-only \
```

```suggestion
   && apt-get purge -y --autoremove python3-pip build-essential python3-dev python3-pip automake pkg-config libffi-dev libssl-dev \
```

these packages are not necessary at runtime so can be removed to save space.

(feel free to ignore; I can make a separate PR for that)

---

I can only agree. Install dependencies optimization deserves a PR on its own imo, we could further optimize layers with a more granular script.

---

- #1800